### PR TITLE
Issue 12224 integration test urlmap vanityurl

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java [hunk 1]
```diff
@@ -279,6 +282,73 @@ public void redirectHomePageTest() throws IOException, DotDataException {
...
+     * @throws DotDataException
+     */
+    @Test
+    public void vanityRedirectURLMAP() throws IOException, DotDataException {
+
+        Contentlet vanityURLContentlet = null;
+
+        try {
+
+            List<ContentType> listContentType = APILocator.getContentTypeAPI(user).findUrlMapped();
```
Comment (jdotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java]: +1

File: dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java [hunk 2]
```diff
@@ -279,6 +282,73 @@ public void redirectHomePageTest() throws IOException, DotDataException {
...
+            ContentType newsContentType =  null;
+            for(ContentType ct : listContentType){
```
Comment (jdotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java]: and use a better name such as contentType

File: dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java [hunk 3]
```diff
@@ -279,6 +282,73 @@ public void redirectHomePageTest() throws IOException, DotDataException {
...
+                if(("news").equalsIgnoreCase(ct.variable())){
+                    newsContentType = ct;
+                    break;
+                }
+            }
+            final Contentlet urlmapContentlet = contentletAPI.findByStructure(new StructureTransformer(newsContentType).asStructure(), user, true, -1, 0).get(0);
+
+            String forwardTo = "/news/" + urlmapContentlet.getStringProperty("urlTitle");
```
Comment (jdotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java]: +1

File: dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java [hunk 4]
```diff
@@ -279,6 +282,73 @@ public void redirectHomePageTest() throws IOException, DotDataException {
...
+                            "/urlnews",
+                            forwardTo, 200, 1, defaultLanguageId);
+            filtersUtil.publishVanityUrl(vanityURLContentlet);
+
+            VanityURLFilter filter = new VanityURLFilter();
+            FilterChain chain = Mockito.mock(FilterChain.class);
+
+            Logger.info(this.getClass(),
+                    "/urlnews should forward to " + forwardTo);
+            HttpServletRequest request = getMockRequest("demo.dotcms.com", "/urlnews");
```
Comment (jdotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java]: +1

**Review by jdotcms — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**